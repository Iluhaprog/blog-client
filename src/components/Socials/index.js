import soc from './Socials';
import { connect } from 'react-redux'
import { setFooterSocialLinksFetch } from '../../actoins/socialLink';

const mapStateToProps = state => ({
    links: state.socialLink.footerLinks,
});
const mapDispatchToProps = dispatch => ({
    getLinks: () => {
        dispatch(setFooterSocialLinksFetch());
    }
});

const Socials = connect(mapStateToProps, mapDispatchToProps)(soc);

export { Socials };