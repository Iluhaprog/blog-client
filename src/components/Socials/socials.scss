.socials {
    &__link {
        margin: 5px 10px;

        img {
            object-fit: contain;
            object-position: center;
        }
    }
}