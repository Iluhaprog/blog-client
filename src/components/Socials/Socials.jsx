import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Row } from '../containers';
import './socials.scss';

function Socials(props) {
    const fileApi = process.env.REACT_APP_FILES_URL;

    useEffect(() => {
        props.getLinks();
    }, [])

    return (
        <div className='socials'>
            <Row justifyContent='c' alignItems='c'>
                { 
                    props.links.map(link => (
                        <a 
                            key={link.id + link.link + Date.now()}
                            href={link.link} 
                            className='socials__link'
                        >
                            { 
                                link.preview 
                                    ? <img 
                                        width={40}
                                        height={40}
                                        src={fileApi + link.preview} 
                                        alt='link'
                                    />
                                    : link.title
                            }
                        </a>
                    )) 
                }
            </Row>
        </div>
    );
}

Socials.defaultProps = {
    links: [],
    getLinks: () => {},
};

Socials.propTypes = {
    links: PropTypes.array,
    getLinks: PropTypes.func,
};

export default Socials;