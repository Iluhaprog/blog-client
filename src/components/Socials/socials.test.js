import React from 'react';
import { render } from '@testing-library/react';
import Socials from './Socials';

test('Should render socials component', () => {
    const { asFragment } = render(<Socials />);
    expect(asFragment()).toMatchSnapshot();
});


test('Should render socials component with links', () => {
    const { asFragment } = render(
        <Socials 
            links={[
                {title: 'title', preview: 'preview', link: 'test'},
                {title: 'test', link: 'link'},
            ]}/>
    );
    expect(asFragment()).toMatchSnapshot();
});