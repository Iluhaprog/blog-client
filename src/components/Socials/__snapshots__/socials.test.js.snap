// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render socials component 1`] = `
<DocumentFragment>
  <div
    class="socials"
  >
    <div
      class="row row_jc-c row_ai-c "
    />
  </div>
</DocumentFragment>
`;

exports[`Should render socials component with links 1`] = `
<DocumentFragment>
  <div
    class="socials"
  >
    <div
      class="row row_jc-c row_ai-c "
    >
      <a
        class="socials__link"
        href="test"
      >
        <img
          alt="link"
          height="40"
          src="http://localhost:3001/file/get/preview"
          width="40"
        />
      </a>
      <a
        class="socials__link"
        href="link"
      >
        test
      </a>
    </div>
  </div>
</DocumentFragment>
`;
